spec/test_repositories.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module TestRepositories
+  module Strategy
+    class One
+      def find(number, unscoped: true)
+        puts "number: #{number} unscoped: #{unscoped.inspect}"
+        number
+      end
+    end
+  end
+
+  module Wrapper
+    class Add5ToNumber
+      def find(number, unscoped:)
+        yield(number + 5, unscoped: unscoped)
+      end
```

Comment:
this somehow is complementary to the below wrappers with before_<method_name> + after_<method_name>

Reply:
you would do it like this, e.g. for sending an update message:

```ruby
let(:add5_wrapper) do
  Class.new do
    def update(id:, changeset:)
      before = Repositories::Foo.find(id: id)
      after = yield(id: id, changeset: changeset)
      Messages.publish(build_update_message(before, after)
    end
  end
end
```

Reply:
its before and after in one. yield gives you the "after"